### DISCUSSION TITLE
[UI Design ideas] Quickly switch logbook and location

### DISCUSSION THREAD

**Discussion by ei8fdb** [magicbug/Cloudlog#3217] [Ideas / Feature Requests]
I regularly operate from different locations (mainly from parks, and other cities). Sometimes it's easy to forget to change the logbook/location.

Here are 2 simple UI design ideas to make changing Logbook and/or Location easier.

Does anyone else operate from different locations regularly and so need to change Logbook/Location often? Do you forget to change?
 
### Idea 1
**Animation showing UI design idea**
![idea-1-swap-logbook-location](https://github.com/user-attachments/assets/6b9497fd-3b05-4212-aeba-49fe71a81dd2)

**Text description**
- User goes to ovewrview page `/index.php/logbook`
- `Logbook: Name`, and `Location: Name` fields are dropdown fields
- User clicks Logbook name dropdown and chooses from list of created Locations
- User clicks Location name dropdown and chooses from list of created Locations
- When ready, user clicks `change button`

### Idea 2
**Animation showing UI design idea**
![idea-2-swap-logbook-location](https://github.com/user-attachments/assets/4c972c29-9e94-42fb-b5c3-0cea0854607a)

**Text description of UI**
- User goes to ovewrview page `/index.php/logbook`
- Beside the current `Logbook: Name`, and `Location: Name` fields, there is a `change button`
- User clicks `change button` and modal opens
- Modal shows a table of the logbooks user has created
- User can choose a different logbook by clicking relevant radio button
- When ready, user clicks `next button`, at botton of modal
- It shows a table of all locations user has created
- User can choose a different location
- When ready, user clicks `Set active button`
- User is returned to main UI with Logbook and Location now updated to their preference

**Comment by magicbug** [comment]
I think you just won the best feature request, a quick change would be a fantastic idea and I especially like idea 2

**Reply by ei8fdb** [reply]
Excellent! If it helps I'll open an issue and add the individual designs with annotations.

I was thinking about the need for a search field in case there were a lot of Logbooks, or locations but left it out.

It'd be helpful to get an idea from people here how many locations/logbooks they have configured.

If you're a Cloudlog user and you're interested in this feature, I'd appreciate if you help me by doing a quick 1 question survey - https://github.com/magicbug/Cloudlog/discussions/3219

telling me how many locations and logbooks you've got configured.

**Comment by magicbug** [comment]
Definitely needs a search option, but I'll work on it this week and see what can be done

**Reply by ei8fdb** [reply]
@magicbug I'll add that search to it then create an issue.
